cluster/gce/windows/common.psm1

```diff
@@ -544,5 +545,16 @@ function Test-IsTestCluster {
   return $false
 }
 
+# Returns true if this node uses a TPM to bootstrap its authentication to the
+# master. $KubeEnv is a hash table containing the kube-env metadata
+# keys+values.
+function Test-NodeUsesTpmBootstrap {
+  param (
+    [parameter(Mandatory=$true)] [hashtable]$KubeEnv
+  )
+
+  return $KubeEnv.Contains('TPM_BOOTSTRAP_CERT')
```

Comment:
is this exactly the same as how linux node checks whether tpm should be enabled?

not sure if there is validation/unittest already to ensure this won't be accidentally broken.

Reply:
Updated to make this test the same as what's used on Linux: https://github.com/kubernetes/kubernetes/blob/8af6906d1f300ff739ac1a846d4108d134412103/cluster/gce/gci/configure.sh#L289. I'll also add a GKE test for this dependency.

It might be safer to test for the presence of a TPM here; something like https://www.top-password.com/blog/find-out-if-your-windows-pc-has-a-tpm-chip/ might work, but I'm not sure how reliable / future-proof that would be. And I'm not sure that a TPM is necessary anyway for plugin-based authentication. I updated a bunch of function names and comments to be a little more general about this.